frontend/packages/kubevirt-plugin/integration-tests/tests/cdi-upload.scenario.ts

```diff
@@ -66,28 +74,102 @@ describe('KubeVirt CDI Upload', () => {
-      await browser.wait(until.textToBePresentInElement(cdiUploadView.uploadProgress, '100%'));
-      await click(cdiUploadView.viewStatusID);
-      await browser.wait(until.textToBePresentInElement(pvcStatus, PVC_STATUS.BOUND));
-    },
-    CDI_UPLOAD_TIMEOUT_SECS,
-  );
+    it('ID(CNV-5176) It shows an error when uploading data to golden OS again', async () => {
+      await uploadForm.openForm();
+      await uploadForm.selectGoldenOS(rhel7PVC.os);
+      await browser.wait(until.presenceOf(errorMessage));
+      expect(errorMessage.getText()).toContain('Operating system source already defined');
+    });
+  });
+
+  describe('KubeVirt GOLDEN OS Creation', () => {
+    const leakedResources = new Set<string>();
+    const fedoraPVC = new PVC(FEDORA_PVC);
+    const win10PVC = new PVC(WIN10_PVC);
+
+    it(
+      'ID(CNV-5042) Create multiple VMs from the same golden os template',
+      async () => {
+        const vm1 = new VMBuilder(getBasicVMBuilder())
+          .setOS(rhel7PVC.os)
+          .generateNameForPrefix('auto-clone-vm1')
+          .setStartOnCreation(true)
+          .build();
+
+        const vm2 = new VMBuilder(getBasicVMBuilder())
+          .setOS(rhel7PVC.os)
+          .generateNameForPrefix('auto-clone-vm2')
+          .setStartOnCreation(true)
+          .build();
+
+        await rhel7PVC.create();
+        await withResource(leakedResources, vm1.asResource(), async () => {
+          await vm1.create();
+          await vm1.navigateToDetail();
+        });
+
+        await withResource(leakedResources, vm2.asResource(), async () => {
+          await vm2.create();
+          await vm2.navigateToDetail();
+        });
+        await rhel7PVC.delete();
+      },
+      VM_BOOTUP_TIMEOUT_SECS + CLONE_VM_TIMEOUT_SECS,
+    );
+
+    it(
+      'ID(CNV-5041) VM can be up after deleting backend golden PVC',
+      async () => {
+        const fedora = new VMBuilder(getBasicVMBuilder())
+          .setOS(fedoraPVC.os)
+          .generateNameForPrefix('auto-clone-vm-with-pvc-deleted')
+          .setStartOnCreation(false)
+          .build();
+
+        await fedoraPVC.create();
+        await fedora.create();
+        await fedora.waitForStatus(VM_STATUS.Off);
+        await fedoraPVC.delete();
+        await withResource(leakedResources, fedora.asResource(), async () => {
+          await fedora.start();
+          await fedora.navigateToDetail();
+        });
+      },
+      VM_BOOTUP_TIMEOUT_SECS + CLONE_VM_TIMEOUT_SECS,
+    );
+
+    it(
+      'ID(CNV-5043) Create Fedora/RHEL/Windows VMs from golden os template',
+      async () => {
+        // skip creating fedora/rhel vm here as it's covered above.
+        const win10 = new VMBuilder(getBasicVMBuilder())
+          .setOS(win10PVC.os)
+          .setFlavor(flavorConfigs.Medium)
+          .generateNameForPrefix('auto-clone-win10-vm')
+          .setStartOnCreation(true)
+          .build();
+
+        await win10PVC.create();
```

Comment:
As deleting PVC via kubectl is not work, it can be done by adding a datavolume and asscoiate the PVC name/namespace to it, put this DV into removeLeakableResource, wdyt?

Reply:
yes if the PVC is backed by DV, then we should remove the DV

Reply:
done, added a method to get DV resource from pvc instance.